pkg/cmd/cli/cmd/newapp.go

```diff
@@ -511,6 +521,20 @@ func setLabels(labels map[string]string, result *newcmd.AppResult, ignoreFailure
 	return nil
 }
 
+func hasLabel(labels map[string]string, result *newcmd.AppResult) (bool, error) {
+	for _, obj := range result.List.Items {
+		objCopy, err := kapi.Scheme.DeepCopy(obj)
+		if err != nil {
+			return false, err
+		}
+		err = util.AddObjectLabelsWithFlags(objCopy.(runtime.Object), labels, util.ErrorOnExistingDstKey)
```

Comment:
because we don't care for how we're using this.  all we care is whether the label exists or not, not what value it has.

if the label exists, we don't want to override it, end of story.  finding out it exists but has the same value we intend to set doesn't have any benefit.